Meta/Lagom/Tools/CodeGenerators/LibWeb/GenerateAriaRoles.cpp

```diff
@@ -0,0 +1,278 @@
+        VERIFY(value.is_object());
+        JsonObject const& value_object = value.as_object();
+
+        auto class_definition_generator = TRY(generator.fork());
+        class_definition_generator.set("spec_link"sv, value_object.get_deprecated_string("specLink"sv).value());
+        class_definition_generator.set("description"sv, value_object.get_deprecated_string("description"sv).value());
+        class_definition_generator.set("name"sv, name);
+        class_definition_generator.append(R"~~~(
+// @spec_link@
+// @description@
+class @name@ :
+)~~~");
+
+        JsonArray const& super_classes = value_object.get_array("superClassRoles"sv).value();
+        bool first = true;
+        TRY(super_classes.try_for_each([&](JsonValue const& value) -> ErrorOr<void> {
+            VERIFY(value.is_string());
+
+            class_definition_generator.append(first ? " "sv : ", "sv);
+            class_definition_generator.append(TRY(String::formatted("public {}", value.as_string())));
+            first = false;
+            return {};
+        }));
+
+        class_definition_generator.append(R"~~~(
+{
+public:
+    @name@(AriaData const&);
+
+    virtual HashTable<StateAndProperties> supported_states() const override;
+    virtual HashTable<StateAndProperties> supported_properties() const override;
+
+    virtual HashTable<StateAndProperties> required_states() const override;
+    virtual HashTable<StateAndProperties> required_properties() const override;
+
+    virtual HashTable<StateAndProperties> prohibited_properties() const override;
+    virtual HashTable<StateAndProperties> prohibited_states() const override;
+
+    virtual HashTable<Role> required_context_roles() const override;
+    virtual HashTable<Role> required_owned_elements() const override;
+    virtual bool accessible_name_required() const override;
+    virtual bool children_are_presentational() const override;
+    virtual DefaultValueType default_value_for_property_or_state(StateAndProperties) const override;
+protected:
+    @name@();
+)~~~");
+
+        auto name_from_source = value.as_object().get("nameFromSource"sv).value();
+        if (!name_from_source.is_null())
+            class_definition_generator.append(R"~~~(
+public:
+    virtual NameFromSource name_from_source() const override;
+)~~~");
+        class_definition_generator.appendln("};");
+        return {};
+    }));
+
+    generator.appendln("}");
+
+    TRY(file.write_until_depleted((generator.as_string_view().bytes())));
+    return {};
+}
+
+ErrorOr<String> generate_hash_table_population(JsonArray const& values, StringView hash_table_name, StringView enum_class)
+{
+    StringBuilder builder;
+    TRY(values.try_for_each([&](auto& value) -> ErrorOr<void> {
+        VERIFY(value.is_string());
+        TRY(builder.try_appendff("    {}.set({}::{});\n", hash_table_name, enum_class, value.as_string()));
+        return {};
+    }));
```

Comment:
Before populating the table, we can call `HashTable::ensure_capacity()`, which will only resize the table once, instead of possibly multiple times.